flaskbb/plugins/models.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+"""
+    flaskbb.plugins.models
+    ~~~~~~~~~~~~~~~~~~~~~~~
+
+    This module provides registration and a basic DB backed key-value
+    store for plugins.
+
+    :copyright: (c) 2017 by the FlaskBB Team.
+    :license: BSD, see LICENSE for more details.
+"""
+from sqlalchemy import UniqueConstraint
+from sqlalchemy.ext.associationproxy import association_proxy
+from sqlalchemy.orm.collections import attribute_mapped_collection
+
+from flaskbb.extensions import db
+from flaskbb.utils.database import CRUDMixin
+
+
+class PluginStore(CRUDMixin, db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    key = db.Column(db.Unicode(255), nullable=False)
+    value = db.Column(db.PickleType, nullable=False)
+    # Enum?
```

Comment:
Sounds good, do we go the Enum34 route or just an older SQLA enum (think it's either a list of strings or a space separated string)

Reply:
I'd go for Enum34 and change the data type to [ChoiceType](https://sqlalchemy-utils.readthedocs.io/en/latest/data_types.html#module-sqlalchemy_utils.types.choice) as we have ``sqlalchemy-utils`` already as dependency.